nf_core/modules/test_yml_builder.py

```diff
@@ -227,20 +227,22 @@ def create_test_file_dict(self, results_dir, is_repeat=False):
         test_files = []
         for root, dir, file in os.walk(results_dir):
             for elem in file:
-                elem = os.path.join(root, elem)
-                test_file = {"path": elem}  # add the key here so that it comes first in the dict
-                # Check that this isn't an empty file
-                if self.check_if_empty_file(elem):
-                    if not is_repeat:
-                        self.errors.append(f"Empty file found! '{os.path.basename(elem)}'")
-                # Add the md5 anyway, linting should fail later and can be manually removed if needed.
-                #  Originally we skipped this if empty, but then it's too easy to miss the warning.
-                #  Equally, if a file is legitimately empty we don't want to prevent this from working.
-                elem_md5 = self._md5(elem)
-                test_file["md5sum"] = elem_md5
-                # Switch out the results directory path with the expected 'output' directory
-                test_file["path"] = elem.replace(results_dir, "output")
-                test_files.append(test_file)
+                # Check that the file is not versions.yml
+                if elem != "versions.yml":
+                    elem = os.path.join(root, elem)
+                    test_file = {"path": elem}  # add the key here so that it comes first in the dict
+                    # Check that this isn't an empty file
+                    if self.check_if_empty_file(elem):
+                        if not is_repeat:
+                            self.errors.append(f"Empty file found! '{os.path.basename(elem)}'")
+                    # Add the md5 anyway, linting should fail later and can be manually removed if needed.
+                    #  Originally we skipped this if empty, but then it's too easy to miss the warning.
+                    #  Equally, if a file is legitimately empty we don't want to prevent this from working.
+                    elem_md5 = self._md5(elem)
+                    test_file["md5sum"] = elem_md5
+                    # Switch out the results directory path with the expected 'output' directory
+                    test_file["path"] = elem.replace(results_dir, "output")
+                    test_files.append(test_file)
```

Comment:
Very minor, but I'd usually write this with a `continue` instead of a huge `if` block. Less indentation = easier to read, plus a smaller diff.

```suggestion
                # Check that the file is not versions.yml
                if elem = "versions.yml":
                    continue

                elem = os.path.join(root, elem)
                test_file = {"path": elem}  # add the key here so that it comes first in the dict
                # Check that this isn't an empty file
                if self.check_if_empty_file(elem):
                    if not is_repeat:
                        self.errors.append(f"Empty file found! '{os.path.basename(elem)}'")
                # Add the md5 anyway, linting should fail later and can be manually removed if needed.
                #  Originally we skipped this if empty, but then it's too easy to miss the warning.
                #  Equally, if a file is legitimately empty we don't want to prevent this from working.
                elem_md5 = self._md5(elem)
                test_file["md5sum"] = elem_md5
                # Switch out the results directory path with the expected 'output' directory
                test_file["path"] = elem.replace(results_dir, "output")
                test_files.append(test_file)
```